### PR TITLE
Added ConfigM's

### LINKED ISSUE #882: Add defaultConfigs to the keymaps again.
When the keymaps got split, the defaultConfigs were lost. They need to be re-added before #873 can be solved.

### PR DESCRIPTION
**Pull request by noughtmare**
Fixes #882 

`Yi.Config.Default.JavaScriptMode` is a bit long to type, but I couldn't think of anything else.

### CODE REVIEW

File: yi-misc-modes/src/Yi/Config/Default/MiscModes.hs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module Yi.Config.Default.MiscModes (configureMiscModes) where
+
+import Lens.Micro.Platform ((%=))
+import Yi.Config.Simple    (ConfigM)
+import Yi.Config.Lens      (modeTableA)
+import Yi.Modes
+import Yi.Types            (AnyMode (..), Mode)
+
+configureMiscModes :: ConfigM ()
+configureMiscModes = do
```
Comment (jhance) [inline yi-misc-modes/src/Yi/Config/Default/MiscModes.hs]: `mapM_ configureMode [ ... ]`

**Comment by noughtmare** [comment]
using `mapM_` gives this error:
```
src/Yi/Config/Default/MiscModes.hs:19:5-13:
        Couldn't match type ‘Yi.Lexer.OCaml.Token’
                       with ‘Yi.Style.UIStyle -> Yi.Style.Style’
        Expected type: Mode
                         (Yi.Syntax.OnlineTree.Tree (Yi.Lexer.Alex.Tok Yi.Style.StyleName))
          Actual type: Yi.Mode.Common.TokenBasedMode Yi.Lexer.OCaml.Token
        In the expression: ocamlMode
        In the second argument of ‘mapM_’, namely
          ‘[cMode, objectiveCMode, cppMode, cabalMode, ....]’
```

File: yi-misc-modes/src/Yi/Config/Default/MiscModes.hs (hunk 1 above)
Comment (noughtmare) [inline yi-misc-modes/src/Yi/Config/Default/MiscModes.hs]: The modes are not all the same type and thus cannot be put in a list.

Comment (jhance) [inline yi-misc-modes/src/Yi/Config/Default/MiscModes.hs]: `AnyMode` them?

**Pull request merged**